Fixes problem with OpenLDAP+OpenSSL cannot trust intermediate
or leaf certificates. 
Patch was developed in-house; it is Solaris specific and
will not be contributed upstream.

diff --git a/libraries/libldap/tls_o.c b/libraries/libldap/tls_o.c
index 3f2813f..ca3df3a 100644
--- a/libraries/libldap/tls_o.c
+++ b/libraries/libldap/tls_o.c
@@ -467,6 +467,16 @@ tlso_ctx_init( struct ldapoptions *lo, struct ldaptls *lt, int is_server )
 		}
 	}
 #endif
+
+#ifdef X509_V_FLAG_PARTIAL_CHAIN
+	/*
+	 * Allow intermediate or leaf certificates in the trust list to
+	 * act as trust anchors.
+	 */
+	X509_STORE_set_flags(SSL_CTX_get_cert_store(ctx),
+	    X509_V_FLAG_PARTIAL_CHAIN);
+#endif
+
 	return 0;
 }
 
